chain/chain-primitives/src/error.rs

```diff
@@ -13,6 +13,27 @@ use near_primitives::serialize::to_base;
 use near_primitives::sharding::{ChunkHash, ShardChunkHeader};
 use near_primitives::types::{BlockHeight, ShardId};
 
+#[derive(thiserror::Error, Debug)]
+pub enum QueryError {
+    #[error("Account ID #{requested_account_id} is invalid")]
+    InvalidAccount { requested_account_id: near_primitives::types::AccountId },
+    #[error("Account ID #{requested_account_id} does not exist while viewing")]
+    AccountDoesNotExist { requested_account_id: near_primitives::types::AccountId },
+    #[error("Contract ID #{contract_account_id} code does not exist while viewing")]
+    ContractCodeDoesNotExist { contract_account_id: near_primitives::types::AccountId },
+    #[error("Access key for public key #{public_key} does not exist while viewing")]
+    AccessKeyDoesNotExist { public_key: near_crypto::PublicKey },
+    #[error("Storage error occurred: #{storage_error:?}")]
+    StorageError {
+        #[from]
+        storage_error: near_primitives::errors::StorageError,
+    },
```

Comment:
I feel like it should be merged into the `InternalError` variant. I feel uneasy about merging a perfectly structured variant into a string, but it seems that after all the `StorageError` is implemented as an internal error representation anyway.